copy/copy.go

```diff
@@ -1253,6 +1253,7 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, to
 			logrus.Debugf("Failed to retrieve partial blob: %v", err)
 			return false, types.BlobInfo{}
 		}(); reused {
+			ic.c.printCopyInfo("blob", srcInfo)
```

Comment:
Isn’t this too late? I understand the point of of “copying…” to be to show what the code _will_ be doing afterwards = which part is taking too long. Printing this only after a successful copy is not what that means.

Reply:
- A very minimal local improvement would be to change the message to be “Trying to use a partial pull…” just so that the two aren’t duplicated. OTOH that’s clutter that is, most of the time, not useful.
- Really, after `TryReusingBlob…` fails, we know we are going to copy a layer, so we could `Printf` at that point, instead of tying that to progress bars. (A notable downside is that this would make the two output kinds less connected).
- Ideally, I think `private.ImageDestination` should have an `LayerAllowsPutBlobPartial` check (which does the annotation comparisons from `c/storage/pkg/chunked.GetDiffer`); `copyLayer` would then call that first, and not bother with the progress bars, or any user-visible activity, at all, for blobs where this isn’t an option. (Combined with the message change to “Trying to use a partial pull”.)

Reply:
I do not know another way.  As far as I understood the code, the bar would just get removed if it fails but we cannot revert the "copying ..." line once it's out.

I figured it's less ugly to have a line later than twice.

Reply:
The updated version should address the problem.  In the `writer == io.Discard` case we never really cared whether we skipped the copy or not, so I moved it out of the progress bar creation and made it explicit for copying a config and a layer.